### PR TITLE
tests: speed up tests with batch put and mem disk

### PR DESCRIPTION
**Pull request by BusyJay**
### What problem does this PR solve?

There are many heavy write in raftstore tests, I have seen some write IO
task takes several seconds to finished, which will certainly make most
test cases timeout. This PR utilizes memory disk to speed up those
operations.
    
Ref #11177

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: tests/integrations/raftstore/test_conf_change.rs [hunk 1]
```diff
@@ -605,26 +604,8 @@ fn test_transfer_leader_safe<T: Simulator>(cluster: &mut Cluster<T>) {
...
-        );
-    }
 
-    // Test transfer leader after a safe duration.
-    thread::sleep(cfg.raft_store.raft_reject_transfer_leader_duration.into());
-    assert_eq!(
-        cluster.leader_of_region(region_id).unwrap().get_id(),
-        leader_id
-    );
+    // Any uptodate follower can become leader.
```
Comment (gengliqi) [inline tests/integrations/raftstore/test_conf_change.rs:608]:
```suggestion
    // Any up-to-date follower can become leader.
```
```suggestion
    // Any uptodate follower can become leader.
```

File: components/test_util/src/lib.rs [hunk 2]
```diff
@@ -93,3 +93,50 @@ pub fn alloc_port() -> u16 {
...
+        }
+    }
+}
+
+/// Memory disk is provided for CI. By default, all raftstore test cases
+/// are run on memory disk. But in some cases, testing IO stats from system
+/// for example, you may want to run cases on physical disk instead.
+///
+/// The correctness of this API depends on running test case one by one.
+pub fn guarantee_on_physical_disk() -> GuaranteeOnPhysicalDisk {
```
Comment (NingLin-P) [inline components/test_util/src/lib.rs:133]: Could we add some config (like `try_men_disk`) to control using what kind of disk instead? And setting the default value based on `#[cfg(any(test))]`, if `try_men_disk` is true then call `temp_dir_in_mem` in `test_raftstore`.

**Review by gengliqi — APPROVED** [review]
LGTM

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/encryption/src/encrypted_file/mod.rs [hunk 3]
```diff
@@ -69,8 +69,7 @@ impl<'a> EncryptedFile<'a> {
         let mut tmp_file = OpenOptions::new()
             .create(true)
             .write(true)
-            .open(&tmp_path)
-            .unwrap_or_else(|e| panic!("EncryptedFile::write {:?}: {}", &tmp_path.to_str(), e));
+            .open(&tmp_path)?;
```
Comment (gregwebs) [inline components/encryption/src/encrypted_file/mod.rs:72]: It looks like we have lost a very useful error message.

File: components/encryption/src/manager/mod.rs [hunk 4]
```diff
@@ -691,14 +691,12 @@ impl DataKeyManager {
...
-        self.rotate_terminal
-            .send(())
-            .expect("DataKeyManager drop send");
-        self.background_worker
-            .take()
-            .expect("DataKeyManager worker take")
-            .join()
-            .expect("DataKeyManager worker join");
+        if let Err(e) = self.rotate_terminal.send(()) {
+            info!("failed to terminate background rotation, are we shutting down?"; "err" => %e);
```
Comment (gregwebs) [inline components/encryption/src/manager/mod.rs:695]: Previously this was a panic? Shouldn't it at least be logged at the error level?

Comment (BusyJay) [reply components/encryption/src/manager/mod.rs:695]: Yes, it's panic in the past. This happens when shutting down, so may not be an error at all.

File: components/encryption/src/encrypted_file/mod.rs (hunk 3 above)
Comment (BusyJay) [reply components/encryption/src/encrypted_file/mod.rs:72]: Good point, let me send a new PR to log the path.

**Pull request merged**